getTextColor now works for grayscale values

quetzaluz opened the pull request:
- Fixed an index lookup error for grayscale values
- Ensured grayscale values can be parsed and returned as hex code; added new spec
- Added spec coverage for text styles (bold, italic) as well

arasabbasi commented:
erm... can you remove jspdf.debug.js and jspdf.min.js from your pull request please? Those files are generated automatically and merging them is causing problems.

quetzaluz commented:
Files removed per request. We might want to update this part of the README to make it clear that they shouldn't be committed
```
## Contributing
Build the library with `npm run build`. This will fetch all dependencies and then compile the `dist` files. To see the examples locally you can start a web server with `npm start` and go to `localhost:8000`.
```

Definitely won't commit those files going forward!

arasabbasi commented:
dont worry, everything is ok

arasabbasi commented on jspdf.js:
```diff
@@ -1118,15 +1118,16 @@ var jsPDF = (function(global) {
-            x = Math.floor(parseFloat(colorEncoded[i]) * 255).toString(16);
-            colorAsHex += (x.length == 1) ? "0"+x : x;
-          }
+        if (colorEncoded.length == 2 && colorEncoded[colorEncoded.length-1] == 'g') {
```
` if (colorEncoded.length == 2 && colorEncoded[colorEncoded.length-1] == 'g') {`

Well umm... if colorEncoded.length equals 2, then colorEncoded.length -1 equals 1

So why not instead?

` if (colorEncoded.length === 2 && colorEncoded[1] == 'g') {`

quetzaluz replied:
Good point, will revise

arasabbasi commented on jspdf.js:
```diff
@@ -1118,15 +1118,16 @@ var jsPDF = (function(global) {
+        var x;
+        var colorAsHex = '#'
+        for (var i = 0; i < 3; i++) {
+          x = Math.floor(parseFloat(colorEncoded[i]) * 255).toString(16);
```
Probably better to merge these two lines to 

`colorAsHex += padd2(Math.floor(parseFloat(colorEncoded[i]) * 255).toString(16));`

quetzaluz replied:
Looks like padd2 won't work because hex codes can be alphanumeric and padd2 calls parseInt:
```
    padd2 = function padd2(number) {
      return ('0' + parseInt(number)).slice(-2);
    },
```

I'll still work on making this a one-liner though, I agree it's a good idea

arasabbasi commented on jspdf.js:
```diff
@@ -1118,15 +1118,16 @@ var jsPDF = (function(global) {
+          var floatVal = parseFloat(colorEncoded[0])
+          colorEncoded = [floatVal, floatVal, floatVal, 'r']
+        }
+        var x;
```
remove variable declaration if you would use proposed change below.

arasabbasi commented on jspdf.js:
```diff
@@ -1118,15 +1118,16 @@ var jsPDF = (function(global) {
+        var colorAsHex = '#'
+        for (var i = 0; i < 3; i++) {
+          x = Math.floor(parseFloat(colorEncoded[i]) * 255).toString(16);
+          colorAsHex += (x.length == 1) ? "0"+x : x;
```
remove line, if applying change proposed above

arasabbasi commented:
My mistake with padd2. I think your PR is merge-ready.

@MrRio 
I think, it is ready.

arasabbasi commented:
@quetzaluz 
One last thing. Not important, but maybe for the style...
Could you please use === instead of == and use semicolons in jspdf.js? 

semicolons in the test-files are imho not "importent".

We will someday start linting the js-files, and doing it now, would be less work in the future.

But I repeat: Great work :)

quetzaluz commented:
Updated to use `===` and semicolons appropriately; squashed this into the previous commit because I don't want to spam your repo with commits over a little feature.

arasabbasi commented:
Perfect!
